src/redhat-jboss-eap.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from bs4 import BeautifulSoup
+from common import dates, http, releasedata
+
+"""Fetches RedHat JBoss EAP releases and versions"""
+
+VERSION_URLS = [
+    "https://docs.redhat.com/en/documentation/red_hat_jboss_enterprise_application_platform/7.4/",
+    "https://docs.redhat.com/en/documentation/red_hat_jboss_enterprise_application_platform/8.0/"
+]
+
+RELEASES_INFO = "https://access.redhat.com/product-life-cycles/api/v1/products?name=Red%20Hat%20JBoss%20Enterprise%20Application%20Platform"
+
+# Given a dict, find inside the list of phases the one that matches the phase_name, extract and parse it's associated date
+def find_date_field(elem, phase_name):
+    entry = list(filter(lambda it: it.get("name") == phase_name, elem["phases"]))[0]
+    return dates.parse_datetime(entry["date"])
+
+with releasedata.ProductData("redhat-jboss-eap") as product_data:
+    # Versions
+    for response in http.fetch_urls(VERSION_URLS):
+        soup = BeautifulSoup(response.text, features="html5lib")
+        # Find all links inside a H3 heading
+        links = [heading.find("a") for (i, heading) in enumerate(soup.find_all("h3")) if heading.find("a")]
+        # From those links, keep the ones pointing to access.redhat.com containing the text " Update "
+        links = [link for link in links if link.get("href").startswith("https://access.redhat.com/articles") and link.getText().find(" Update ") > 0]
+
+        # For every link, get the last modified date of the associated knowledge base article and create a version entry
+        for link in links:
+            name = link.getText().strip()
+            # Clean up the name to obtain an identifier in the form X.Y.Z[.W]
+            identifier = name.replace(" Update", "").replace(" 0", ".").replace(" ", ".")
+            article_url = link.get("href")
+            article_response = http.fetch_url(article_url)
+            article_soup = BeautifulSoup(article_response.text, features="html5lib")
+            date_header = article_soup.find(class_="header")
```

Comment:
This will contain the last update to the page, not the release date. Not sure we should use this date.

Reply:
That's fair.

I found that the release date information can be found in these pages:

* [JBoss EAP 7 Maintenance Schedule](https://access.redhat.com/articles/2332721)
* [JBoss EAP 8 Maintenance Schedule](https://access.redhat.com/articles/7039745)

The first link can be visited without an account, but the second link for JBoss EAP 8 requires a RedHat account 😞

Reply:
For JBoss 8, there's info in their Maven repo metadata for JBoss 8: https://[maven.repository.redhat.com/ga/org/jboss/eap/channels/eap-8.0/maven-metadata.xml](https://maven.repository.redhat.com/ga/org/jboss/eap/channels/eap-8.0/maven-metadata.xml) which contains only the date of the most recent update, which is probably enough.